src/partition.rs

```diff
@@ -108,6 +108,34 @@ impl BlockSize {
   const BLOCK_SIZE_HEIGHT_LOG2: [usize; BlockSize::BLOCK_SIZES_ALL] =
     [2, 3, 2, 3, 4, 3, 4, 5, 4, 5, 6, 5, 6, 7, 6, 7, 4, 2, 5, 3, 6, 4];
 
+  pub fn from_width_and_height(w: usize, h: usize) -> BlockSize {
+    match (w, h) {
+      (4, 4) => BLOCK_4X4,
+      (4, 8) => BLOCK_4X8,
+      (8, 4) => BLOCK_8X4,
+      (8, 8) => BLOCK_8X8,
+      (8, 16) => BLOCK_8X16,
+      (16, 8) => BLOCK_16X8,
+      (16, 16) => BLOCK_16X16,
+      (16, 32) => BLOCK_16X32,
+      (32, 16) => BLOCK_32X16,
+      (32, 32) => BLOCK_32X32,
+      (32, 64) => BLOCK_32X64,
+      (64, 32) => BLOCK_64X32,
+      (64, 64) => BLOCK_64X64,
+      (64, 128) => BLOCK_64X128,
+      (128, 64) => BLOCK_128X64,
+      (128, 128) => BLOCK_128X128,
+      (4, 16) => BLOCK_4X16,
+      (16, 4) => BLOCK_16X4,
+      (8, 32) => BLOCK_8X32,
+      (32, 8) => BLOCK_32X8,
+      (16, 64) => BLOCK_16X64,
+      (64, 16) => BLOCK_64X16,
+      _ => BLOCK_INVALID
```

Comment:
when it can happen? I'd mark it as `unreachable!()`

Reply:
Good point. Still learning rust :-)